### DISCUSSION TITLE
Multi-language, how Craft determines the CopyFrom after adding website and Section

### DISCUSSION THREAD

**Discussion by eugenebroeren** [craftcms/cms#8534] [Ideas]
### Description
Problem when introducing a new website. Sections are filled (i think with the first alphabetic language).
I will not copy from the default language.  

It would be nice if i can choose what the prefferred copyfrom language is (when activating the section for that language). At that  point (activate the section and press Save) CraftCMS copies the content from 
website.at to website.zz.

I don't want website.at data to website.zz but for example website.fr to website.zz

website.at websiteAt de-AT https://website.at/ website.eu
website.com websiteCom en-US http://website-comdev.nl/ website.eu
website.de websiteDe de-DE https://website.de/ website.eu
website.eu websiteEu en V https://website.eu/ website.eu
website.fr websiteFr fr https://website.fr/ website.eu
website.nl websiteNl nl-NL https://website.nl/ website.eu
website.sk websiteSk sk-SK https://www.website.sk/ website.eu

### Steps to reproduce

1. add a new website (example website.zz)
2. activate a section for website.zz 
3. the section data for website.zz is now filled with data from website.at

### Additional info

- Craft version: Craft Pro 3.0.2
- PHP version: 7.1.16
- Database driver & version: MySQL 5.5.56

**Comment by brandonkelly** [comment]
It will get copied from the primary site, if the section is enabled for that site. Otherwise, it will just go with the first site that it can find which the section already was (and still is) enabled for.

**Comment by eugenebroeren** [comment]
Thanks, that looked logical but i did a test and the primary is ignored (it is enabled for that section).
It just copies the first (website.at)

So normally this would work for adding a new website (say i want french)
- Add new website
- Set website.fr primary
- Enable the section for the new website

**Comment by brandonkelly** [comment]
Doh you’re right, that was an unintentional regression. Fixed for the next release, but I’m leaving this issue open because I do agree it would be nice to be able to choose which site the new sites should be duplicated from.

**Comment by eugenebroeren** [comment]
Hello Brandon,

>  Doh you’re right, that was an unintentional regression. Fixed for the
> next release, but I’m leaving this issue open because I do agree it would
> be nice to be able to choose which site the new sites should be duplicated
> from.

Are there plans for fixing the (multilanguage) thing. A "duplicate from"
switch in the section environment.
If we introduce a new language, it is now stuffed with the first language
(we concluded in revious topic).

Is there another way for example if we introduce "Morocco" that it fills
with french, the script what is fired when switching on a section after
adding a new language.

Grtz Eugène

2018-04-17 13:11 GMT+02:00 Brandon Kelly <notifications@github.com>:
